### PR TITLE
Feature granttype password

### PR DESCRIPTION
**Pull request by uhavin**
* Added id_token to the authentication response
* Return 403 rather than 400 on authentication failure

### CODE REVIEW

**Comment by bracki** [comment]
@juanifioren What's the progress here? Is anything missing to merge the PR?

**Comment by juanifioren** [comment]
@wojtek-fliposports can you please review this?. This is important.

**Comment by wojtek-fliposports** [comment]
@juanifioren @bracki reviewing it just now

File: oidc_provider/tests/test_token_endpoint.py [hunk 1]
```diff
@@ -128,6 +136,125 @@ def _get_userinfo(self, access_token):
 
         return userinfo(request)
 
+    def _auth_header(self):
```
Comment (wojtek-fliposports) [inline oidc_provider/tests/test_token_endpoint.py]: please rename function to more Password Auth Header related

File: oidc_provider/tests/test_token_endpoint.py [hunk 2]
```diff
@@ -128,6 +136,125 @@ def _get_userinfo(self, access_token):
...
+        invalid_post = self._password_grant_post_data()
+        invalid_post['password'] = 'wrong!'
+
+        response = self._post_request(
+            post_data=invalid_post,
+            extras=self._auth_header()
+        )
+
+        response_dict = json.loads(response.content.decode('utf-8'))
+        print(response_dict)
```
Comment (wojtek-fliposports) [inline oidc_provider/tests/test_token_endpoint.py]: remove print or change it for logger

File: oidc_provider/tests/test_token_endpoint.py [hunk 3]
```diff
@@ -128,6 +136,125 @@ def _get_userinfo(self, access_token):
...
+        mock_uuid.uuid4.return_value = mock_uuid4
+
+        response = self._post_request(
+            post_data=self._password_grant_post_data(),
+            extras=self._auth_header()
+        )
+
+        response_dict = json.loads(response.content.decode('utf-8'))
+        id_token = JWS().verify_compact(response_dict['id_token'].encode('utf-8'), self._get_keys())
+        print(id_token)
```
Comment (wojtek-fliposports) [inline oidc_provider/tests/test_token_endpoint.py]: remove print or change it for logger

**Comment by bracki** [comment]
@wojtek-fliposports Don't you think we should rather merge this and than apply the changes you requested? Or @uhavin can you implement the changes that have been requested?

**Comment by uhavin** [comment]
@bracki @wojtek-fliposports, Thanks for reviewing this. These are small issues and won't take that much time to resolve. I will be able to fix these today.

**Pull request merged**